notifications/src/main/kotlin/com/amazon/opendistroforelasticsearch/commons/notifications/model/NotificationStatus.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ */
+
+package com.amazon.opendistroforelasticsearch.commons.notifications.model
+
+import com.amazon.opendistroforelasticsearch.notifications.util.fieldIfNotNull
+import com.amazon.opendistroforelasticsearch.notifications.util.logger
+import com.amazon.opendistroforelasticsearch.notifications.util.objectList
+import com.amazon.opendistroforelasticsearch.notifications.util.valueOf
+import org.elasticsearch.common.Strings
+import org.elasticsearch.common.io.stream.StreamInput
+import org.elasticsearch.common.io.stream.StreamOutput
+import org.elasticsearch.common.io.stream.Writeable
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParserUtils
+
+/**
+ * Data class representing Notification Status.
+ */
+data class NotificationStatus(
+    val configId: String,
+    val configName: String,
+    val configType: NotificationConfig.ConfigType,
+    val emailRecipientStatus: List<EmailRecipientStatus> = listOf(),
+    val deliveryStatus: DeliveryStatus? = null
+) : Writeable, ToXContent {
+
+    init {
+        require(!Strings.isNullOrEmpty(configId)) { "config id is null or empty" }
+        require(!Strings.isNullOrEmpty(configName)) { "config name is null or empty" }
+        when (configType) {
+            NotificationConfig.ConfigType.Chime -> requireNotNull(deliveryStatus)
+            NotificationConfig.ConfigType.Webhook -> requireNotNull(deliveryStatus)
+            NotificationConfig.ConfigType.Slack -> requireNotNull(deliveryStatus)
+            NotificationConfig.ConfigType.Email -> require(emailRecipientStatus.isNotEmpty())
```

Comment:
may be we can use deliveryStatus for overall email delivery status?

Reply:
sure